test/test_segmentation_upload.py

```diff
@@ -0,0 +1,31 @@
+from pathlib import Path
+import pytest
+import os
+from omerocrate.uploader import OmeroUploader, ApiUploader, SegmentationUploader, OmeNgffUploader
+from omerocrate.taskqueue.upload import TaskqueueUploader
+from omero.gateway import BlitzGateway
+from util import requires_flower, check_seg_dataset
+
+
+@pytest.mark.skipif(
+    os.getenv("RUNNER_ENVIRONMENT") == "github-hosted",
+    reason="Skip on GitHub-hosted runners due to ROI tool requirement, only run on self-hosted"
+)
```

Comment:
Only the `OmeNgffUploader` should have that skip mark, not the whole test. You can apply it using `pytest.param` like I have done for the `TaskqueueUploader`.

Reply:
I implemented this in the end. As `SegmentationUploader` can now be None I made checking of the ROI conditional.